packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/SimulationEventService.java

```diff
@@ -151,11 +155,30 @@ private void onScimlAllInstanceReceive(final Message message) {
 
 	@RabbitListener(queues = "${terarium.simulation-status}", concurrency = "1")
 	private void onPyciemssOneInstanceReceives(final Message message, final Channel channel) throws IOException {
-		final CiemssStatusUpdate update = ClientEventService.decodeMessage(message, CiemssStatusUpdate.class);
-		if (update == null) {
+		// Parse message to get message type:
+		final String updateType = (ClientEventService.decodeMessage(message, JsonNode.class)).get("type").asText();
+		if (updateType.equals(CiemssStatusType.OPTIMIZE.toString())) {
```

Comment:
FYI this `CiemssStatusType` enum isn't very useful, because you are converting everything to a string at the end anyway. It is functionally equivalent to a class with static string fields but needs more steps.